docs: Typography migration guide

talkor opened the pull request:
https://monday.monday.com/boards/3532714909/pulses/5506419848

shlomitc commented on src/storybook/stand-alone-documentaion/typography/typography-migration-guide.stories.mdx:
```diff
@@ -0,0 +1,106 @@
+import { TEXT, HEADING, EDITABLE_HEADING } from "../../components/related-components/component-description-map";
+import { migrationTable } from "./assets";
+
+<Meta title="Text/Typography Migration Guide" />
```
I think it should be on the root level of the documentation, below the changelog

shlomitc commented on src/storybook/stand-alone-documentaion/typography/typography-migration-guide.stories.mdx:
```diff
@@ -0,0 +1,106 @@
+
+<img src={migrationTable} alt="Migration table" />
+
+### What if I need a variant that doesn't exist?
```
The title and the content aren't related.
For the title - If the variant you need doesn't exist - contact the design system theme for support.
For the content - this is relevant for cases of styling when components can't be used. For example, styling an input element

shlomitc reviewed (changes requested):
Cool
1. Add a link to the migration guide from the `LegacyHeading` and `LegacyEditableHeading` components.
2. for some reason links are opening on a new tab. I thought that was supposed to be fixed already

talkor commented:
> Cool
> 
> 1. Add a link to the migration guide from the `LegacyHeading` and `LegacyEditableHeading` components.
> 2. for some reason links are opening on a new tab. I thought that was supposed to be fixed already

Where would you put a link in the legacy components? We have the DeprecatedWarning component that gets a fixed text, I can make it more flexible for that purpose. WDYT?
Regarding the links I think @SergeyRoyt worked on that, not sure, will check

shlomitc commented:
> Where would you put a link in the legacy components? We have the DeprecatedWarning component that gets a fixed text

sounds good

talkor commented:
> > Where would you put a link in the legacy components? We have the DeprecatedWarning component that gets a fixed text
> 
> sounds good

Done. Regarding the links that open in a new tab, I added a target so they now open in the same tab, though the storybook app refreshes and it supposed to do client side routing, we should use this [addon](https://www.npmjs.com/package/@storybook/addon-links) fro internal navigation, it will make it much smoother. I added a [task](https://monday.monday.com/boards/3532714909/pulses/5540093456) and will make it in a different PR

shlomitc reviewed (approved):
🤩
